src/github.com/matrix-org/dendron/dendron/main.go

```diff
@@ -148,49 +125,62 @@ func main() {
 		}
 	}
 
-	var synapse *exec.Cmd
-	var pusher *exec.Cmd
-	var synchrotron *exec.Cmd
-
 	var synapseLog = log.WithField("synapse", synapseURL.String())
 
+	// Used to terminate dendron.
+	terminate := make(chan string, 1)
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		s := <-signals
+		terminate <- fmt.Sprint("Got signal", s)
+	}()
+
 	if *startSynapse {
-		synapse = exec.Command(*synapsePython, "-m", "synapse.app.homeserver", "-c", *synapseConfig)
+		synapse := exec.Command(*synapsePython, "-m", "synapse.app.homeserver", "-c", *synapseConfig)
 		synapse.Stderr = os.Stderr
 		synapseLog.Print("Starting synapse")
 
 		synapse.Start()
-
-		sh := signalHander{
-			synapseLog: synapseLog,
-			synapse:    synapse.Process,
-		}
-		channel := make(chan os.Signal, 1)
-		signal.Notify(channel, os.Interrupt)
-		go sh.handleSignal(channel)
+		defer synapse.Process.Signal(os.Interrupt)
 
 		if err := waitForSynapse(synapseURL, synapseLog); err != nil {
```

Comment:
This is too easy to confuse with `synapse.Process.Wait()`. Be clearer which is waiting for Synapse to spin up, and which is waiting for the Synapse process to die.